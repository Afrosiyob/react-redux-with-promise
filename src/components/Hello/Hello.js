import React from "react";
import PropTypes from "prop-types";

const Hello = (props) => {
  const { name } = props;

  if (name) {
    return <h3 data-testid="hello-1"> Hello, {name} </h3>;
  } else {
    return <span data-testid="hello-1"> Hi, stranger </span>;
  }
};

Hello.propTypes = {
  name: PropTypes.string,
};

export default Hello;
