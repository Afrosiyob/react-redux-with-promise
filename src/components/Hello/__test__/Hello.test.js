import { cleanup, render, screen } from "@testing-library/react";
import Hello from "../Hello";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("Hello no props", () => {
  render(<Hello />);
  const helloelement = screen.getByTestId("hello-1");
  expect(helloelement).toBeInTheDocument();
  expect(helloelement).toHaveTextContent("Hi, stranger");
  expect(helloelement).toContainHTML(
    '<span data-testid="hello-1"> Hi, stranger </span>'
  );
});

test("Hello with props", () => {
  render(<Hello name="Jane" />);
  const helloelement = screen.getByTestId("hello-1");
  expect(helloelement).toBeInTheDocument();

  expect(helloelement).toHaveTextContent("Hello, Jane");
  expect(helloelement).toContainHTML(
    '<h3 data-testid="hello-1"> Hello, Jane </h3>'
  );
});

test("Hello snapshot", () => {
  const tree = renderer.create(<Hello />).toJSON();

  expect(tree).toMatchSnapshot();

  console.log(tree);
});
