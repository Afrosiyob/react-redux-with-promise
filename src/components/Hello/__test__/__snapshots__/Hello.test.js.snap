// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello snapshot 1`] = `
<span
  data-testid="hello-1"
>
   Hi, stranger 
</span>
`;
